gslib/tests/test_hmac.py

```diff
@@ -112,24 +123,53 @@ def CreateHelper(self, service_account):
 
   def test_malformed_commands(self):
     params = [
-        ('hmac create', 'requires a service account'),
+        ('hmac create', 'requires a service account',
+         'argument SERVICE_ACCOUNT: Must be specified'),
+        ('hmac create -p proj', 'requires a service account',
+         'argument SERVICE_ACCOUNT: Must be specified'),
+        ('hmac delete', 'requires an Access ID',
+         'argument ACCESS_ID: Must be specified'),
+        ('hmac delete -p proj', 'requires an Access ID',
+         'argument ACCESS_ID: Must be specified'),
+        ('hmac get', 'requires an Access ID',
+         'argument ACCESS_ID: Must be specified'),
+        ('hmac get -p proj', 'requires an Access ID',
+         'argument ACCESS_ID: Must be specified'),
+        ('hmac list account1', 'unexpected arguments',
+         'unrecognized arguments'),
+        ('hmac update keyname', 'state flag must be supplied',
+         'Exactly one of (--activate | --deactivate) must be specified.'),
+        ('hmac update -s INACTIVE', 'requires an Access ID',
+         'argument ACCESS_ID: Must be specified'),
+        ('hmac update -s INACTIVE -p proj', 'requires an Access ID',
+         'argument ACCESS_ID: Must be specified'),
+    ]
+    for command, gsutil_error_substr, gcloud_error_substr in params:
+      expected_status = 2 if self._use_gcloud_storage else 1
+
+      stderr = self.RunGsUtil(command.split(),
+                              return_stderr=True,
+                              expected_status=(expected_status))
+
+      if self._use_gcloud_storage:
+        self.assertIn(gcloud_error_substr, stderr)
+      else:
+        self.assertIn(gsutil_error_substr, stderr)
+
+  def test_malformed_commands_that_cannot_be_translated_using_the_shim(self):
+    if self._use_gcloud_storage:
+      raise unittest.SkipTest('These commands cannot be translated using the'
```

Comment:
Can you check around if we have a decorator for this?

Reply:
sure

Reply:
No, We don't any decorator for this.